<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="dabadav • 20-Mar-2022" />


<title>Docking and macromolecular complexes</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />
<link id="font-awesome-1-attachment" rel="attachment" href="site_libs/font-awesome-5.1.0/fonts/fontawesome-webfont.ttf"/>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="assets/lab.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
#TOC {
  display: none !important;
}
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"><img src="assets/logo.svg" id="logo" style="height:20px;margin:0;"/></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="home_schedule.html">Schedule</a>
</li>
<li>
  <a href="home_content.html">Content</a>
</li>
<li>
  <a href="home_precourse.html">Precourse</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Docking and macromolecular complexes</h1>
<h3 class="subtitle">Structural Bioinformatics</h3>
<h4 class="author"><b>dabadav</b> • 20-Mar-2022</h4>

</div>


<!-- rmd lab header -->
<p><br></p>
<p>Docking is the <strong>prediction the structure of complexes</strong> such as:</p>
<ul>
<li>Protein-Ligand, Protein-Protein, Protein-NA, NA-Ligand, NA-NA</li>
</ul>
<p>In order for these complexes to be formed, their integrants need to recognize each other in a <strong>selective</strong> and dynamic way, this is achieved by structural complementarity.</p>
<p>This prediction might be <strong>useful to</strong>:</p>
<ul>
<li>Understanding interactions, roles of specific amino acids, design of mutations and changes of activity</li>
<li>Comparison of affinities of different molecules</li>
<li>Drug design</li>
</ul>
<div id="molecular-docking-types" class="section level1">
<h1><span class="header-section-number">1</span> Molecular Docking types</h1>
<p>Docking approaches have to deal with the conformational changes between unbound and bound structures.</p>
<p><strong>Bound vs Unbound</strong></p>
<ul>
<li>Bound: goal is to reproduce a known complex where the starting coordinates of the individual molecules are taken from the crystal of the complex</li>
<li>Unbound: significantly more difficult problem, the starting coordinates are taken from the unbound molecules</li>
</ul>
<p>Early algorithms in order to be faster assumed that docked molecules do not change conformations, treating them as rigid bodies.</p>
<p><strong>Flexible vs Rigid</strong></p>
<ul>
<li>Rigid: procedures that perform rigid body search</li>
<li>Felxible: procedures that consider possible conformational changes (protein flexibility)</li>
</ul>
<p>Whether the binding site of molecules is known or not the search space will vary. Global docking ismore demanding in terms of computational time and the results are less accurate.</p>
<p><strong>Global vs Local</strong></p>
<ul>
<li>Global: search for the location of the binding site and a search to figure out the exact orientation of the ligand in the binding site.</li>
<li>Local: if binding site is known, we just need to orient the ligand in the binding site.</li>
</ul>
</div>
<div id="docking-strategy" class="section level1">
<h1><span class="header-section-number">2</span> Docking Strategy</h1>
<div class="pull-left-50">
<ul>
<li><p>Protein representation</p></li>
<li><p>Search method</p></li>
<li><p>Scoring method</p></li>
</ul>
<p>Scoring might be done during the search process, or a posteriori.</p>
<ul>
<li>Structural complementarity
<ul>
<li>Robust, low resolution</li>
</ul></li>
<li>Classical force-fields, Statistical Potentials
<ul>
<li>High resolution</li>
<li>Easy to transfer</li>
</ul></li>
<li>Empirical functions
<ul>
<li>DGbind obtained from function fitted to experimental data</li>
<li>Based on vdW, electrostatic, hydrogen bonds, desolvation, enthropy, hydrophobicy, …</li>
<li>Reproducing affinity of ligation</li>
</ul></li>
<li>Refinement?
<ul>
<li>Molecular Dynamics, …</li>
</ul></li>
</ul>
<div class="pull-right-50 text-center">

</div>
<p><img src="assets/strategy.png" style="width:90%;" class="fancyimage"/></p>
</div>
<p><br></br></p>
</div>
<div id="ligand-protein-docking" class="section level1">
<h1><span class="header-section-number">3</span> Ligand-Protein Docking</h1>
<p>Small-molecule docking</p>
<ul>
<li><strong>Molecular Docking</strong>
<ul>
<li>Prediction of 3D structure of ligand-protein or protein-protein complexes.</li>
<li><em>One receptor – one or few ligands</em></li>
<li>Quality of the structure is the main objective. Realistic binding energies</li>
<li>Usually combined with other techniques, as MD</li>
<li>Experimental information can be considered</li>
</ul></li>
<li><strong>Virtual screening</strong>
<ul>
<li>Identification of possible ligands from compound databases</li>
<li><em>One receptor – multiple ligands (&gt; 10^6)</em></li>
<li>Calculation should be fast (&gt;10000 ligand-receptor dockings / day / proc.)</li>
<li>The main objective is to select “some” ligands, that can be optimized with other methods</li>
</ul></li>
<li><strong>Reverse Screening</strong>
<ul>
<li>Identification of possible receptors for a known ligand</li>
<li><em>One ligand - multiple receptors</em></li>
<li>Points to possible side effects</li>
</ul></li>
<li><strong>Active site prediction</strong>
<ul>
<li>Identification of binding regions</li>
</ul></li>
</ul>
<p>• Complete atomic representation
- Large cost, high resolution</p>
<p>• Simplified representations
- Quick and robust. Low resolution.</p>
<p>• 3D Grid representations (receptor)
- Easier energy calculation
- Definition of “pharmacophores” (MIPs)</p>
<p>• Flexibility (Ligand and receptor)
- Ensembl docking</p>
</div>
<div id="protein-docking" class="section level1">
<h1><span class="header-section-number">4</span> Protein docking</h1>
<p>The approaches to protein-protein docking have a lot in common with small molecule docking</p>
<p>Still based on the combination of search algorithm coupled to a scoring function</p>
<p>Scoring functions are essentially the same (since we are still dealing with atomic interactions), however the major problem is that the conformational space we now need to search is massive</p>
<ul>
<li>Large systems: contact area: 1500-3000 A2</li>
<li>Large num of degrees of freedom + Large conformational flexibility</li>
<li>Few “easy” interactions
<ul>
<li>Hydrophobic contacts</li>
<li>Average 1 H-bond / 170A^2</li>
<li>1 water / 100 A2</li>
</ul></li>
</ul>
<div id="pure-ab-initio-docking" class="section level2">
<h2><span class="header-section-number">4.1</span> “Pure” Ab initio docking</h2>
<ul>
<li>Only information about ligand and receptor structures is known
<ul>
<li>Pseudo-random approaches (simulation, optimization)</li>
<li>Directed search (Geometric hashing)</li>
<li>Brute-Force approaches (Grid-based, FFT)</li>
</ul></li>
</ul>
<p>ADDITIONAL TRICKS</p>
<p><strong>Flexible docking</strong></p>
<ul>
<li><p>Traditional MD</p>
<ul>
<li>Refine structures after docking</li>
</ul></li>
<li><p>Conformational sampling</p>
<ul>
<li>Rigid docking with set of possible conformations (experimental)</li>
<li>Conformational search added to position and orientation (MC)</li>
</ul></li>
<li><p>Combined cycles of docking and simulation</p></li>
<li><p>RosettaDock combines rigid body MonteCarlo orientation/traslation + Montecarlo amon other rotamer libraries (very expensive)</p></li>
</ul>
<p><strong>Soft docking</strong></p>
<ul>
<li><p>Backbone is still rigid</p></li>
<li><p>Sidechain flexibility is mimicked using “soft” VdW potentials and/or coarser FFT grids</p></li>
</ul>
</div>
<div id="data-driven-docking-template-based" class="section level2">
<h2><span class="header-section-number">4.2</span> Data driven docking (template based)</h2>
<ul>
<li><p>Experimental, homology data is used</p>
<ul>
<li>Template based, using data from homologues</li>
</ul></li>
<li><p>Co-evolution methods (AlphaFold)</p>
<ul>
<li>Protein structure prediction</li>
<li>Uses data from “massive” MSA (Multiple Seq Alignment) and correlated mutations to define correlated positions &gt; 3D contacts (predicted contacts become restrains)</li>
</ul>
<p><img src="assets/coevolution.png" style="width:100%;" class="fancyimage"/></p></li>
<li><p>Interface prediction</p>
<ul>
<li>Reveal interfaces without structure prediction</li>
</ul></li>
</ul>
</div>
</div>
<div id="conclusion" class="section level1">
<h1><span class="header-section-number">5</span> Conclusion</h1>
<ul>
<li>Protein docking works
<ul>
<li>(Much less efficient than ligand docking)</li>
</ul></li>
<li>Lots of methods exists, no clear winner</li>
<li>Data-driven methods can generate better models if data is available</li>
<li>Flexibility, conformational changes are the major problems</li>
<li>Interface and interaction predictions (without docking) are possible and useful</li>
</ul>
</div>


<footer class="footer">
<div class="container footer-container">
<div class="row">

<div class="footer-left col-sm-6">
<p class="small">
<b>2021</b> • dabadav
</p>
</div>

<div class="footer-right col-sm-6">
<p>
<span>
    <span class="footericon" style="padding-right:4px; padding-left:4px">
      <a href="https://twitter.com/xiluum"><svg style="height:0.8em;top:.04em;position:relative;fill:#bdbdbd;" viewBox="0 0 512 512"><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"></path>
    </svg>
    </a>
    </span>
    <span class="footericon" style="padding-left:4px">
      <a href="https://www.linkedin.com/in/dante-avi%C3%B1%C3%B3-mart%C3%ADnez-3a0485229/">
        <svg style="height:0.8em;top:.04em;position:relative;fill:#bdbdbd;" viewBox="0 0 448 512"><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"></path>
        </svg>
    </a>
    </span>
  </span>
</p>
</div>

</div>
</div>
</footer>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
